src/main/java/de/tum/in/www1/artemis/service/metis/ReactionService.java

```diff
@@ -78,6 +87,10 @@ public Reaction createReaction(Long courseId, Reaction reaction) {
             AnswerPost answerPost = answerPostService.findById(posting.getId());
             reaction.setAnswerPost(answerPost);
             reaction.setUser(user);
+            // check if the user already reacted with the reaction to add (-> avoid duplicated reactions) before saving
+            if (reactionRepository.findReactionsByAnswerPostIdAndEmojiIdAndUserId(answerPost.getId(), reaction.getEmojiId(), user.getId()).size() > 0) {
+                throw new BadRequestAlertException("You can only react once with a certain emoji", METIS_REACTION_ENTITY_NAME, "duplicatedReaction");
+            }
```

Comment:
This seems like a repeated throw clause. Maybe it can be converted to a **orElseThrow** function like we usually do, since I've seen [them ](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server/?highlight=orelsethrow#avoid-service-dependencies)to be way more preferred. 
_ReactionRepository.java_
  ```
  default List<Reaction> findReactionsByPostIdAndEmojiIdAndUserIdElseThrow(Long postId, String emojiId, Long userId) throws EntityNotFoundException {
        return findReactionsByPostIdAndEmojiIdAndUserId(postId, emojiId, userId).stream().findAny().orElseThrow((() -> new EntityNotFoundException("Reactions", postId));
    }
```

then 

```suggestion
            reactionRepository.findReactionsByPostIdAndEmojiIdAndUserIdElseThrow(answerPost.getId(), reaction.getEmojiId(), user.getId())
```

Reply:
The thing is, that it should throw an error _if_ there is an entity found, so the reaction would be duplicated (same user, same posting, same reaction). What you are suggesting is the other way around, throw an error if we cannot find the reaction we are searching for.

Reply:
ah I see, then sth like this maybe, but that's up to you if you find it useful for your case or not.
```
    default void findNoReactionsByPostIdAndEmojiIdAndUserIdElseThrow(Long postId, String emojiId, Long userId) throws BadRequestAlertException {
        findReactionsByPostIdAndEmojiIdAndUserId(postId, emojiId, userId).stream().findAny().ifPresent(s -> { throw new BadRequestAlertException("You can only react once with a certain emoji", emojiId, "duplicatedReaction");});
    }
```

Reply:
yep, that worked, I did it the way you proposed